test/test_content_type.py

```diff
@@ -0,0 +1,219 @@
+        return 'service'
+
+    @classmethod
+    def catalog_config(cls) -> dict[CatalogName, config.Catalog]:
+        return {
+            cls.catalog: config.Catalog(name=cls.catalog,
+                                        atlas='hca',
+                                        internal=False,
+                                        plugins=dict(metadata=config.Catalog.Plugin(name='hca'),
+                                                     repository=config.Catalog.Plugin(name='dss')),
+                                        sources=set())
+        }
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        @cls.app_module.app.route('/mode/{mode}')
+        def route(mode):
+            if mode == 'string':
+                return '<script />'
+            elif mode == 'json':
+                return {'<script />': '<iframe></iframe>'}
+            elif mode == 'Response200':
+                return Response(status_code=200,
+                                body='<script />')
+            elif mode == 'Response200textplain':
+                return Response(status_code=200,
+                                headers={'Content-Type': 'text/plain'},
+                                body='<script />')
+            elif mode == 'Response200texthtml':
+                return Response(status_code=200,
+                                headers={'Content-Type': 'text/html'},
+                                body='<script />')
+            elif mode == 'NotFoundError':
+                raise NotFoundError('<script />')
+            elif mode == 'ChaliceUnhandledError':
+                raise ChaliceUnhandledError('<script />')
+            elif mode == 'Exception':
+                raise Exception('<script />')
+            else:
+                assert False
+
+        setattr(cls.app_module, 'foo', route)
+
+    def test_response(self):
+        """
+        Verify the Content-Type and html escaping of the various response types.
+        """
+
+        def shrink_traceback(s: str) -> str:
+            # Keep the first and last line of a traceback, replace the lines
+            # in between with a single '...' line.
+            if s.startswith('Traceback'):
+                lines = s.split('\n')
+                assert lines[-1] == '', s  # since traceback ends with a '\n'
+                s = '\n'.join([lines[0], '...', lines[-2]])
+            elif s.startswith('<html>'):
+                # Assumes traceback is a json dumped string inside <pre></pre> tags
+                pattern = re.compile(r'(<pre>)'
+                                     r'("Traceback.*?\\n)'  # 1st line of traceback
+                                     r'.*\\n'  # middle lines
+                                     r'(.*)\\n'  # last line of traceback
+                                     r'("</pre>)')
+                s = re.sub(pattern, r'\1\2...\\n\3\4', s)
+            return s
+
+        def get_test_values() -> JSON:
```

Comment:
```suggestion
        def get_test_values(debug) -> JSON:
```

That makes it immediately obvious why this is a method. No need to explain anything and no need to patch state examined by the method. But wait with this until we discuss in PL.